.github/workflows/test.yml

```diff
@@ -29,7 +31,7 @@ jobs:
       uses: actions/cache@v1
       with:
         path: target
-        key: ${{ runner.os }}-cargo-build-target-v1-${{ hashFiles('**/Cargo.lock') }}
+        key: ${{ runner.os }}-cargo-build-target-v${{ env.CACHE_GENERATION }}-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Perhaps something to do in a different PR but should this be extended to the other key in that file?

Reply:
I wasn't sure. It'd be easy enough to do, but I admit I don't have much depth when it comes to these caching mechanisms. So I limited to just the key that seemed to be failing.

Do you understand these things? I'm happy to push a patch if you do and think it'd be a good idea. If not, any idea who does?

Reply:
To some extent :) I think we should cover both. They have the same key (based on the output from the Github Actions) so in theory the jobs can share the files. If they can't (because the keys end up being different) and we need to bump the CACHE_GENERATION value, worse case the first time the job runs it will need to download and build everything. For reference, this will happen naturally if the cache file is not accessed in 7 days.

Reply:
Done! (In 2b58192)